src/components/sidebar/index.tsx

```diff
@@ -108,8 +108,11 @@ const Sidebar = ({
     )
     sidebarContent = (
       <ul className={s.navList}>
-        {filteredMenuItems.map((item: FilteredNavItem) => (
-          <SidebarNavMenuItem item={item} key={item.id} />
+        {filteredMenuItems.map((item: FilteredNavItem, index) => (
+          <SidebarNavMenuItem
+            item={item}
+            key={item.id ?? item.title ?? index}
```

Comment:
I think the fix here needs to be making sure there's an `id` on every element, which is important because they're used in the HTML to programmatically connect different elements together. I spiked this out in https://github.com/hashicorp/dev-portal/pull/329/files#diff-85db3c95ee8d187913e75e7bd50242deb7ec7d36a4ed18b739ecfbde99e2af53 and it worked nicely! I forgot about it until now, I can pull it out into a separate PR if that's cool?

Reply:
Ah yeah I think that would be good! FWIW, dividers are one that I'm not sure we'll ever have an `id` for 💭